lib/cretonne/src/ir/dfg.rs

```diff
@@ -682,6 +698,51 @@ impl DataFlowGraph {
                         })
     }
 
+    /// Removes `val` as an `Ebb` argument and transform it into an alias of `alias_val`. This
+    /// means that all the uses of `val` will refer to `alias_val`.
+    ///
+    /// *Important*: to ensure O(1) deletion, this method swaps the removed argument with the
+    /// last `Ebb` argument. This can disrupt all the branch instructions jumping to this
+    /// `Ebb` for which you have to change the jump argument order if necessary.
+    ///
+    /// Panics if `val` is not an `Ebb` argument. Returns `true` if `Ebb` arguments have been
+    /// swapped.
+    pub fn remove_ebb_arg(&mut self, val: Value) -> bool {
```

Comment:
For safety, I think we should call this `swap_remove_ebb_arg`. Just so people know how it behaves.

I think it would be useful to return `num as usize` from this function instead of a bool. That will be useful for a caller who needs to update all of the predecessor branches. They can call `swap_remove` with that index (adjusted for non-EBB args).

This would require implementing `swap_remove` for `EntityList`, of course. That method could also simplify your implementation below:

```rust
self.ebbs[ebb].args.swap_remove(num, &mut self.value_lists);
if let Some(last_arg_val) = self.ebbs[ebb].args.get(num, &self.value_lists) {
       // We update the position of the old last arg.
       if let ValueData::Arg { num: ref mut old_num, .. } = self.values[last_arg_val] {
           *old_num = num;
       } else {
         panic!("{} should be an Ebb argument but is not", last_arg_val);
    }
}
```

In the case where `swap_remove` removed the last element, `get(num, ...)` will return `None`.